### ISSUE TITLE
gzip defaults

### ISSUE THREAD

**Issue by Zertz** [onmodulus/docker-run-static#1]
I understand why you’d want to force gzip on, but it seems odd to provide defaults for an arbitrary directive. Especially since we have to override `gzip_types` anyway because the list is (understandably) incomplete.

https://github.com/onmodulus/docker-run-static/blob/master/conf/nginx.conf#L21

This causes an issue when attempting to share configuration files between nginx instances hosted elsewhere. For example, with this config on Modulus, we have to comment out the first two lines because otherwise nginx complains about duplicate keys and doesn't start.

``` nginx
# gzip.conf

# gzip on;
# gzip_proxied any;
gzip_vary on;
gzip_comp_level 5;
gzip_types application/javascript application/json text/css text/html … ;
```

Basically, it would be helpful to either have the `gzip` directives removed or be able to entirely override them.

As a side note, every deployment causes the following error to appear in the logs:

``` bash
nginx: [alert] could not open error log file: open() "/var/log/nginx/error.log" failed (13: Permission denied)
```

**Comment by InconceivableDuck** [comment]
Awesome feedback!

What if we supported overriding the base nginx config if you deployed a file called `nginx.conf` in the root directory? That way you could override everything.

The error is a tough one to get rid of but doesn't cause any issues. This is because the in-container user (mop) doesn't have write access to /var, but nginx really wants access to it.

**Comment by Zertz** [comment]
That would be perfect. Also, your default `gzip_types` should probably include `application/javascript`

Yeah, the error isn't a big deal. IIRC, that path is hard-coded and has to be changed when compiling the source.
